easybuild/tools/build_log.py

```diff
@@ -103,25 +104,28 @@ def deprecated(self, msg, max_ver):
 
     def nosupport(self, msg, ver):
         """Print error message for no longer supported behaviour, and raise an EasyBuildError."""
-        self.error("NO LONGER SUPPORTED since v%s: %s; see %s for more information" % (ver, msg, DEPRECATED_DOC_URL))
+        msg = "NO LONGER SUPPORTED since v%s: %s; see %s for more information" % (ver, msg, DEPRECATED_DOC_URL)
+        raise EasyBuildError(msg)
 
     def error(self, msg, *args, **kwargs):
         """Print error message and raise an EasyBuildError."""
-        newMsg = "EasyBuild crashed with an error %s: %s" % (self.caller_info(), msg)
-        fancylogger.FancyLogger.error(self, newMsg, *args, **kwargs)
+        new_msg = "EasyBuild crashed with an error %s: %s" % (self.caller_info(), msg)
+        fancylogger.FancyLogger.error(self, new_msg, *args, **kwargs)
         if self.raiseError:
-            raise EasyBuildError(newMsg)
+            self.deprecated("Automatically raising EasybuildError in error() logging method", '3.0')
```

Comment:
no, `log.error(...)` should be replaced with `raise EasyBuildError(...)`, to make Traceback 'cleaner'?